packages/sol/src/verification.ts

```diff
@@ -0,0 +1,408 @@
+import {
+  COUPON_REGEX,
+  EMAIL_REGEX,
+  HTML_REGEX,
+  HTML_TAGS_REGEX,
+  INSTAPAY_REGEX,
+  LETTERS_ONLY_REGEX,
+  MAX_DISCOUNT_VALUE,
+  MAX_FEEDBACK_TEXT_LENGTH,
+  MAX_INTERVIEW_LEVEL,
+  MAX_MESSAGE_TEXT_LENGTH,
+  MAX_NOTE_TEXT_LENGTH,
+  MAX_PASSWORD_LENGTH,
+  MAX_PLAN_ALIAS_LENGTH,
+  MAX_PLAN_DISCOUNT,
+  MAX_PLAN_WEEKLY_MINUTES,
+  MAX_RATING_TEXT_LENGTH,
+  MAX_RATING_VALUE,
+  MAX_RULE_DURATION_MINUTES,
+  MAX_RULE_TITLE_LENGTH,
+  MAX_TUTOR_ABOUT_TEXT_LENGTH,
+  MAX_TUTOR_BIO_TEXT_LENGTH,
+  MAX_TUTOR_NOTICE_DURATION,
+  MAX_USER_AGE,
+  MIN_DISCOUNT_VALUE,
+  MIN_FEEDBACK_TEXT_LENGTH,
+  MIN_INTERVIEW_LEVEL,
+  MIN_MESSAGE_TEXT_LENGTH,
+  MIN_NOTE_TEXT_LENGTH,
+  MIN_PASSWORD_LENGTH,
+  MIN_PLAN_ALIAS_LENGTH,
+  MIN_PLAN_DISCOUNT,
+  MIN_PLAN_WEEKLY_MINUTES,
+  MIN_RATING_TEXT_LENGTH,
+  MIN_RATING_VALUE,
+  MIN_RULE_DURATION_MINUTES,
+  MIN_RULE_TITLE_LENGTH,
+  MIN_TUTOR_NOTICE_DURATION,
+  MIN_USER_AGE,
+  NUMBERS_ONLY_REGEX,
+  PASSWORD_REGEX,
+  PHONE_NUMBER_REGEX,
+  USER_NAME_REGEX,
+} from "@/constants";
+import { Bank, FieldError, WithdrawMethod } from "@litespace/types";
+import { getSafeInnerHtmlText } from "./utils";
+
+export function isValidEmail(email: string): FieldError.InvalidEmail | true {
+  if (!EMAIL_REGEX.test(email)) return FieldError.InvalidEmail;
+  return true;
+}
+
+export function isValidPassword(
+  password: string
+):
+  | FieldError.PasswordTooShort
+  | FieldError.PasswordTooLong
+  | FieldError.PasswordMissingLetters
+  | FieldError.PasswordMissingNumbers
+  | true {
+  if (!NUMBERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingNumbers;
+  if (!LETTERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingLetters;
+  if (password.length > MAX_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (password.length < MIN_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (PASSWORD_REGEX.test(password)) return true;
+  return true;
+}
+
+export function isValidUserName(
+  userName: string
+): FieldError.InvalidUserName | true {
+  if (!USER_NAME_REGEX.test(userName)) return FieldError.InvalidUserName;
```

Comment:
Add more analysis to the username. 
- Incase the username includes numbers or special letters return `FieldError.UserNameSpecialLettersNotAllowed` and `FieldError.UserNameNumbersNotAllowed`
- Incase the username includes English characters return `FieldError.UserNameEnglishLettersNotAllowed`

Returning these errors will help guild the user to the correct answer.